File: Data Structures & Algorithms/jump-game/submission-4.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxIndex = 0
        for i in range(len(nums)):
            maxIndex = max(maxIndex, i+nums[i])
            if maxIndex >= len(nums)-1:
                return True
            while maxIndex < len(nums) and nums[maxIndex] == 0:
                maxIndex-=1
            if maxIndex <= i:
                return False

File: Data Structures & Algorithms/duplicate-integer/submission-4.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int, int> um;
        for (auto num: nums) {
            if(um[num]!=0) return true;
            um[num] = 1;
        }
        return false;
    }
};